ps_facetedsearch.php

```diff
@@ -1503,4 +1513,88 @@ public function getWidgetVariables($hookName, array $configuration)
     {
         return [];
     }
+
+    /**
+     * Install review indexes table
+     */
+    public function rebuildCommentIndexTable()
+    {
+        $this->getDatabase()->execute('DROP TABLE IF EXISTS `' . _DB_PREFIX_ . 'layered_comment_index`');
+
+        $this->getDatabase()->execute('DROP TABLE IF EXISTS `' . _DB_PREFIX_ . 'layered_comment_index_log`');
+
+        $this->getDatabase()->execute(
+            'CREATE TABLE `' . _DB_PREFIX_ . 'layered_comment_index` (
+            `id_product` INT  NOT NULL,
+            `score` INT NOT NULL,
+            `avg_score` FLOAT NOT NULL,
+            PRIMARY KEY (`id_product`),
+            INDEX `score` (`score`),
+            INDEX `avg_score` (`score`)
+            ) ENGINE=' . _MYSQL_ENGINE_ . ' DEFAULT CHARSET=utf8;'
+        );
+
+        $this->getDatabase()->execute(
+            'CREATE TABLE `' . _DB_PREFIX_ . 'layered_comment_index_log` (
+            `id_comment` INT  NOT NULL,
+            `id_product` INT  NOT NULL,
+            `indexed` TINYINT(1) NOT NULL,
+            PRIMARY KEY (`id_comment`)
+            ) ENGINE=' . _MYSQL_ENGINE_ . ' DEFAULT CHARSET=utf8;'
+        );
+    }
+
+    public function indexReviews($full = false)
+    {
+        if ($full) {
+            $validateCondition = '';
+            if (Configuration::get('PRODUCT_COMMENTS_MODERATE')){
+                $validateCondition = ' where validate = 1';
+            }
+
+            $query = 'SELECT avg(`grade`) as avg_grade, sum(`grade`) as sum_grade, `id_product` FROM ' . _DB_PREFIX_ . 'product_comment '. $validateCondition .' group by id_product';
+
+            foreach ($this->getDatabase()->executeS($query) as $comment) {
+                $this->addCommentIndex($comment);
+            }
+
+            $query = 'SELECT `id_product_comment`, `id_product` FROM ' . _DB_PREFIX_ . 'product_comment' . $validateCondition;
+
+            foreach ($this->getDatabase()->executeS($query) as $commentLog) {
+                $this->addCommentIndexLog($commentLog);
+            }
+
+        } else {
+            $query = 'select pc.id_product_comment, pc.id_product, pc.grade from ps_product_comment as pc
+left join ps_layered_comment_index_log as lc
+on pc.id_product_comment = lc.id_comment
+where lc.indexed is null;';
```

Comment:
```suggestion
            $query = 'SELECT pc.id_product_comment, pc.id_product, pc.grade ' .
                'FROM ' . _DB_PREFIX_ . '_product_comment as pc ' .
                'LEFT JOIN ' . _DB_PREFIX_ . '_layered_comment_index_log as lc ' .
                'pc.id_product_comment = lc.id_comment ' .
                'WHERE lc.indexed IS NULL';
```

Easier to read and fix error with unknown table begining with `ps_`  prefix

Reply:
correct thanks.